lib/run_qsim.h

```diff
@@ -123,17 +123,24 @@ struct QSimRunner final {
   }
 
   /**
-   * Runs the given circuit and make the final state available to the caller.
+   * Runs the given circuit and make the final state available to the caller,
+   * recording the result of any intermediate measurements in the circuit.
    * @param param Options for parallelism and logging.
    * @param circuit The circuit to be simulated.
    * @param state As an input parameter, this should contain the initial state
    *   of the system. After a successful run, it will be populated with the
    *   final state of the system.
+   * @param measure_results As an input parameter, this should be empty.
+   *   After a successful run, this will contain all measurements results from
+   *   the run, ordered by time and qubit index.
    * @return True if the simulation completed successfully; false otherwise.
    */
   template <typename Circuit>
-  static bool Run(const Parameter& param, const Circuit& circuit,
-                  typename Simulator::State& state) {
+  static bool Run(
+      const Parameter& param, const Circuit& circuit,
+      typename Simulator::State& state,
+      std::vector<typename Simulator::StateSpace::MeasurementResult>&
```

Comment:
It might make sense to add `using MeasurementResult = typename Simulator::StateSpace::MeasurementResult;` above and replace `typename Simulator::StateSpace::MeasurementResult` with `MeasurementResult` for better formatting.

Reply:
Done. Also updated the other `using` declarations affected by this.